packages/arb-token-bridge-ui/src/util/teleports/fetchEthTeleportsFromSubgraph.ts

```diff
@@ -0,0 +1,111 @@
+import { isTeleport } from '../../token-bridge-sdk/teleport'
+import { hasL1Subgraph } from '../SubgraphUtils'
+import { getAPIBaseUrl, sanitizeQueryParams } from '../index'
+
+export type FetchEthTeleportsFromSubgraphResult = {
+  transactionHash: string
+  timestamp: string
+  blockCreatedAt: string
+  l2Calldata: string
+  value: string
+  sender: string
+  retryableTicketID: string
+  destAddr: string
+  l1ChainId: string
+  l3ChainId: string
+  teleport_type: 'eth'
+}
+
+/**
+ * Fetches initiated ETH Teleports from subgraph in range of [fromBlock, toBlock] and pageParams.
+ *
+ * @param query Query params
+ * @param query.sender Address that initiated the withdrawal
+ * @param query.receiver Address that received the funds
+ * @param query.fromBlock Start at this block number (including)
+ * @param query.toBlock Stop at this block number (including)
+ * @param query.l2ChainId Chain id for the L2 network
+ * @param query.pageSize Fetch these many records from subgraph
+ * @param query.pageNumber Fetch records starting [pageNumber * pageSize] records
+ * @param query.searchString Searches records through the l1TxHash
+ */
+
+export const fetchEthTeleportsFromSubgraph = async ({
+  sender,
+  receiver,
+  fromBlock,
+  toBlock,
+  l1ChainId,
+  l2ChainId,
+  l3ChainId,
+  pageSize = 10,
+  pageNumber = 0,
+  searchString = ''
+}: {
+  sender?: string
+  receiver?: string
+  fromBlock: number
+  toBlock?: number
+  l1ChainId: number
+  l2ChainId: number
+  l3ChainId: number
+  pageSize?: number
+  pageNumber?: number
+  searchString?: string
+}): Promise<FetchEthTeleportsFromSubgraphResult[]> => {
+  if (toBlock && fromBlock >= toBlock) {
+    // if fromBlock > toBlock or both are equal / 0
+    return []
+  }
+
+  const urlParams = new URLSearchParams(
+    sanitizeQueryParams({
+      sender,
+      receiver,
+      fromBlock,
+      toBlock,
+      l1ChainId,
+      l2ChainId,
+      l3ChainId,
+      pageSize,
+      page: pageNumber,
+      search: searchString
+    })
+  )
+
+  if (!hasL1Subgraph(Number(l2ChainId))) {
```

Comment:
Shouldn't that use `hasTeleporterSubgraph `?

Reply:
No we use the existing L1 subgraph - the one which indexes the deposits also indexes retryables, which we use here.